### PR TITLE
Issue 12207: Adding token_type_hint option

### PR DESCRIPTION
**Pull request by julien2424**
Added the token_type_hint option to OpenidConnectClient and socialLogin/oauth2Login.
for #12207

### CODE REVIEW

**Comment by julien2424** [comment]
#build

File: dev/com.ibm.ws.security.social/resources/OSGI-INF/metatype/metatype.xml [hunk 1]
```diff
@@ -188,6 +188,9 @@
             required="true" type="String" ibm:type="password" />
         <AD id="authorizationEndpoint" name="%authorizationEndpoint" description="%authorizationEndpoint.desc"
             required="false" type="String" default="https://github.com/login/oauth/authorize"/>
+        <AD id="introspectionTokenTypeHint" name="internal" description="internal use only" type="String" required="false" default="access_token">
```
Comment (ayoho) [inline dev/com.ibm.ws.security.social/resources/OSGI-INF/metatype/metatype.xml:191]: Looks like this was actually put under the `com.ibm.ws.security.social.github.metatype` metatype, not the `oauth2Login` metatype. We'll have to come back and update this.

**Comment by ayoho** [comment]
#run-libby-bot

**Comment by julien2424** [comment]
#run-libby-bot

**Pull request merged**